### PR TITLE
Bugfix/attack issue one

### PR DESCRIPTION
**Pull request by freecraver**
includes two bugfixes: 

- attack now takes the max 3 attacker limit into consideration (but simulates bigger attacks for optimization)
- fortify now should no longer lead to an endless loop in case no fortification action is possible

### CODE REVIEW

File: src/main/java/agents/leeroy/Leeroy.java [hunk 1]
```diff
@@ -35,9 +35,9 @@ public A computeNextAction(G game, long computationTime, TimeUnit timeUnit) {
         super.setTimers(computationTime, timeUnit);
         log.info("Computing action");
         Risk risk = (Risk) game;
-        setPhase(game.getBoard());
         A nextAction;
         try {
+            setPhase(game.getBoard());
```
Comment (AntonOellerer) [inline src/main/java/agents/leeroy/Leeroy.java:40]: Was this the reason for the strange error?

File: src/main/java/agents/leeroy/FortificationActionSupplier.java [hunk 2]
```diff
@@ -51,10 +51,10 @@
                     .collect(Collectors.toSet()));
             nextLevelTerritoryIds.addAll(subseqTerritorySet);
         }
-        if (! levelActions.isEmpty()) {
+        if (! levelActions.isEmpty() || nextLevelTerritoryIds.isEmpty()) {
+            // some found on level or last level (i.e. no fortification possible)
             return levelActions;
```
Comment (AntonOellerer) [inline src/main/java/agents/leeroy/FortificationActionSupplier.java:56]: What is `levelActions` if `nextLevelTerritoryIds.isEmpty()`?

File: src/main/java/agents/leeroy/Leeroy.java (hunk 1 above)
Comment (freecraver) [reply src/main/java/agents/leeroy/Leeroy.java:40]: I don't think so. The strange error might still be caused by my fortification logic, but I'm not sure. In the last few runs I didn't get any error

**Review by AntonOellerer — APPROVED** [review]

File: src/main/java/agents/leeroy/FortificationActionSupplier.java (hunk 2 above)
Comment (freecraver) [reply src/main/java/agents/leeroy/FortificationActionSupplier.java:56]: it's an empty HashSet, i.e. no actions possible

**Pull request merged**